mesh/mesh.go

```diff
@@ -337,26 +387,17 @@ func (m *Mesh) getTxs(txIds []types.TransactionId, l *types.Layer) []*types.Tran
 }
 
 func (m *Mesh) PushTransactions(l *types.Layer) {
-	validBlockTxs, invalidBlockTxs := m.ExtractUniqueOrderedTransactions(l)
+	validBlockTxs := m.ExtractUniqueOrderedTransactions(l)
 	numFailedTxs, err := m.ApplyTransactions(l.Index(), validBlockTxs)
 	if err != nil {
 		m.With().Error("failed to apply transactions",
 			log.LayerId(l.Index().Uint64()), log.Int("num_failed_txs", numFailedTxs), log.Err(err))
 		// TODO: We want to panic here once we have a way to "remember" that we didn't apply these txs
 		//  e.g. persist the last layer transactions were applied from and use that instead of `oldBase`
 	}
-	m.removeFromUnappliedTxs(validBlockTxs, invalidBlockTxs, l.Index())
-	for _, tx := range invalidBlockTxs {
-		err = m.blockBuilder.ValidateAndAddTxToPool(tx)
-		// We ignore errors here, since they mean that the tx is no longer valid and we shouldn't re-add it
-		if err == nil {
-			m.With().Info("transaction from contextually invalid block re-added to mempool",
-				log.TxId(tx.Id().ShortString()))
-		}
-	}
```

Comment:
Since there could be no contextually invalid blocks in transaction processor anymore, these transactions should not be taken from the mempool at all, making it easier to manage the tx's and not having to remove and add all the time